autonomous/autonomous.py

```diff
@@ -13,12 +13,32 @@
 from controllers.shooter import ShooterController
 from controllers.indexer import IndexerController
 from utilities import trajectory_generator
+from typing import List
+from enum import Enum
 
 
-class AutoBase(AutonomousStateMachine):
-    """Follows a smoothed linear path between waypoints
-    if it has a ball it stops before the next waypoint to fire it, otherwise dosent stop"""
+class WaypointType(Enum):
+    PICKUP = 0
+    SHOOT = 1
+    WAYPOINT = 2
+
+
+@dataclass
+class Waypoint:
+    def __init__(
+        self,
+        x: float,
+        y: float,
+        angle: float,
+        waypoint_type: WaypointType = WaypointType.WAYPOINT,
+    ):
+        """x, y: field position in meters
+        angle: robot angle in degrees"""
```

Comment:
Given we normally use radians in our coordinate system, I'm not sure this is a great idea. Would rather pass in a Rotation2d instead so it's explicit in the waypoint list which units we're using.

Reply:
I think you forgot to fix this doc.

Reply:
```suggestion
        rotation: desired robot heading
        """
```